zntrack/core/node.py

```diff
@@ -75,6 +76,28 @@ def fs(self) -> dvc.api.DVCFileSystem:
                 time.sleep(0.1)
         raise dvc.utils.strictyaml.YAMLValidationError
 
+    @contextlib.contextmanager
+    def patch_open(self) -> typing.ContextManager:
+        """Patch the open function to use the Node's file system."""
+        original_open = open
```

Comment:
Have a `directory` argument that will only patch opening files in that directory.